Hai Reveal conversations: simplify dialog in hai reveal 4 missions A03-A07 and correct a dialog bug (#7571)
endless-sky/endless-sky: data/hai/hai reveal 4 middle.txt

		has "Hai Reveal [A02] Symbol Identification: done"
	on offer
		event "alondo symbol timer" 12
		log `Alondo asked me to check out the Hai spaceports in various systems near the wormhole, to see if anything unexpected is happening.`
		conversation

> 			`You make what is beginning to feel like a standard request to land at the secure spaceport and disembark with relevant copies of files on a data stick in hand. It doesn't take long to track down Alondo, though you do have to wait nearly half an hour while he finishes a meeting with a Hai official. When he's finished, he waves you into the room and you take a seat. The seat you choose is clearly designed to be comfortable for a Hai, making a strange experience for a human.`

Reviewer: ```suggestion
			`You make what is beginning to feel like a standard request to land at the secure spaceport and disembark with relevant copies of files on a data stick in hand. It doesn't take long to track down Alondo, though you do have to wait nearly half an hour while he finishes a meeting with a Hai official. When he's done, he waves you into the room and asks you to take a seat. The seat you choose is clearly designed to be comfortable for a Hai, making for a rather strange experience for a human.`
```

I put a few things back. Removing that little "of some kind" made the repetition more visible.

> 			`You make what is beginning to feel like a standard request to land at the secure spaceport and disembark with relevant copies of files on a data stick in hand. It doesn't take long to track down Alondo, though you do have to wait nearly half an hour while he finishes a meeting with a Hai official. When he's done, he waves you into the room and asks you to take a seat. The seat you choose is clearly designed to be comfortable for a Hai, making for a rather strange experience for a human.`

Reviewer: ```suggestion
			`You make what is beginning to feel like a standard request to land at the secure spaceport and disembark with relevant copies of files on a data stick in hand. It doesn't take long to track down Alondo, though you do have to wait nearly half an hour while he finishes a meeting with a Hai official. When he's done, Alondo waves you into the room and asks you to take a seat. The seat you choose is clearly designed to be comfortable for a Hai, making for a rather strange experience for a human.`
```
It's a small change but there's tons of 'he' in these few sentences.

> 			`	You show Alondo the cat symbol you found, and he says, "There's a research institute on Earth that specializes in history of politics during the early hyperspace travel era. If anyone would recognize this symbol, it's them. I'll pass it on and let you know what they say."`

Reviewer: ```suggestion
			`	You show Alondo the cat symbol you found. He says, "There's a research institute on Earth that specializes in the history of politics during the early hyperspace travel era. If anyone would recognize this symbol, it's them. I'll pass it on and let you know what they say."`
```

> 			`	"In the meantime," he says, "could you stop by some of the Hai worlds near the wormhole? They've obviously ramped up security on Allhome, but I'd like to have a pair of friendly eyes check it out. The Hai may be using this as a cover to put thir military into a threatening position... not that I think they'd do that, but we've just been through one war. I may be getting paranoid sitting in the most Hai place that there is."`

Author: What justification should the player give for discretion?

> 			`	"In the meantime," he says, "could you stop by some of the Hai worlds near the wormhole? They've obviously ramped up security on Allhome, but I'd like to have a pair of friendly eyes check it out. The Hai may be using this as a cover to put thir military into a threatening position... not that I think they'd do that, but we've just been through one war. I may be getting paranoid sitting in the most Hai place that there is."`

Author: I put the comment back in.

		has "Hai Reveal [A02] Symbol Identification: done"
	on offer
		event "alondo symbol timer" 12
		log `Alondo asked me to check out the Hai spaceports in various systems near the wormhole, to see if anything unexpected is happening.`
		conversation
			`You make what is beginning to feel like a standard request to land at the secure spaceport and disembark with relevant copies of files on a data stick in hand. It doesn't take long to track down Alondo, though you do have to wait nearly half an hour while he finishes a meeting with a Hai official. When he's done, Alondo waves you into the room and asks you to take a seat. The seat you choose is clearly designed to be comfortable for a Hai, making for a rather strange experience for a human.`
			# TODO: Replace this paragraph with actual content. Include something about him working at Remington lately.
			`	You show Alondo the cat symbol you found. He says, "There's a research institute on Earth that specializes in the history of politics during the early hyperspace travel era. If anyone would recognize this symbol, it's them. I'll pass it on and let you know what they say."`

> 			`	"In the meantime," he says, "could you stop by some of the Hai worlds near the wormhole? They've obviously ramped up security on Allhome, but I'd like to have a pair of friendly eyes check it out. The Hai may be using this as a cover to put thir military into a threatening position... not that I think they'd do that, but we've just been through one war. I may be getting paranoid sitting in the most Hai place that there is."`

Reviewer 2: ```suggestion
			`	"In the meantime," he says, "could you stop by some of the Hai worlds near the wormhole? They've obviously ramped up security on Allhome, but I'd like to have a pair of friendly eyes check it out. The Hai may be using this as a cover to put their military into a threatening position... not that I think they'd do that, but we've just been through one war. I may be getting paranoid sitting in the most Hai place that there is."`
```

mission "Hai Reveal [A04] Report to Teeneep"
# This mission needs an odd name so it lexically sorts after "Hai Reveal: Pirate Troubles [4]"
# Otherwise, you'll get the Teeneep note about Scar's Legion being resolved,
# before the Scar's Legion issue is resolved
mission "ZZ Hai Reveal [A04] Report to Teeneep"
	landing
	name "Report to Teeneep"
	description `Tell Teeneep about the experience with Scar's Legion.`
	source Hai-home
	source "Hai-home"
	destination Darkwaste
	to offer

> 		has "event: alondo symbol timer"

Author: I think this was an earlier attempt at fixing the problem that required "hr defeated scar's legion".  I will put the symbol timer back in and make sure it works.

> 		has "event: alondo symbol timer"

Author: I reverted this change.

> 			has "hr defeated scar's legion"

Author: It did not trigger without that change. The reason is that if an on offer sets a variable in a conversation (including the ": offered" variable), then that variable is not available to "to offer" conditions in later missions. (Took a while to debug that, and find it in the code.) Hence, I have to trigger on an earlier condition.

	on offer

> 		dialog `Now that the situation with Scar's Legion is resolved, it's time to report to Teeneep about the experience. Maybe it will shed some light on what's going on?`

Author: There is a missing branch in the missions where you "check in with Alondo" instead of Teeneep. You'll get the message to visit Alondo, which you can't do. If things trigger in the wrong order, you're unable to go through the missions.

> 		dialog `Now that the situation with Scar's Legion is resolved, it's time to report to Teeneep about the experience. Maybe it will shed some light on what's going on?`

Author: Actually, scratch that. That problem is unrelated to this line. I'll revert this while I'm testing the timer trigger.

> 		dialog `Now that the situation with Scar's Legion is resolved, it's time to report to Teeneep about the experience. Maybe it will shed some light on what's going on?`

Author: I reverted this change.

	to offer
		has "event: alondo symbol timer"
		or

> 			has "ZZ Hai Reveal [A04] Report to Teeneep: done"
> 			has "Hai Reveal [A04] Report to Teeneep: done"

Reviewer 3: Duplication.
```suggestion
			has "ZZ Hai Reveal [A04] Report to Teeneep: done"
```

Author: That was to avoid breaking old saves during my testing.

			
			label fury
			`	"WHAT!?"`
			`	Teeneep's exclamation is loud enough that every head in the room turns towards you both. "The Elders would prioritize our secrecy over our safety, even when the purpose of the secrecy is to maintain our safety?!"`

> 			`	It takes a few seconds for Teeneep to calm her fury before she continues.`

Author: Her childish behavior in this and A09 is inconsistent with her personality and achievements elsewhere in the campaign. I'm reducing her temper problems to "barely in control of her emotions" rather than "acts like a five-year-old" since it is more plausible.

> 			`	It takes a few seconds for Teeneep to calm her fury before she continues.`

Reviewer 3: In the context of the other discussion about the deals this can be closed off

> 			`	It takes a few seconds for Teeneep to calm her fury before she continues.`

Author: I think you're asking me to delete this line (and the dialog does make sense without it) so I just did that.

		has "Hai Reveal [A05] Take Teeneep to Hai-home: done"
	on offer
		conversation

> 			`As soon as you disembark, Teeneep enters another security checkpoint and shows them some ID, but you're too far to see what it is. A quick inquiry confirms that Alondo is actually around this time and you send him a message to alert him that you're heading his way.`

Reviewer: ```suggestion
			`As soon as you disembark, Teeneep enters another security checkpoint and shows them some ID, but you're too far away to see what it is. A quick inquiry confirms that Alondo is finally around this time, and you send him a message to alert him that you're heading his way.`
```

> 			`	As you approach Alondo's temporary office, you can hear voices through the open door and you slow to a stop as you realize they sound agitated. "Look, Gruyere, you can't expect me to not organize some things around you when you're off being busy elsewhere. Regardless of whether it was your idea."`

Author: It only saved one line on the screen, so I reverted it to the original.

			choice
				`	(Make your appearance.)`
					goto meeting
			label doorway
			`	You wander off for a few minutes, giving Alondo his privacy. Upon your return, you are face to face with Remington who has stood up to greet you.`
			label meeting
			`	"Ah, <last>, please come in." Remington moves over and clears your path to the nearest seat.`
			`	"It's good to see you back again, <last>," says Alondo, "and just in time for something that I think it's best to ask you to do. Would you like to tell the good Captain what the plan is, Remington?"`
			`	The slightly accentuated formality on 'Remington' is not lost on you, but the Syndicate PR man doesn't react to it as he hands you a datapad displaying a hyperspace relay. "Since we met the Hai, the Navy maintained hyperspace relays in Rajak and Hasseleh, linking our communication network to the Hai's through a secure portal. However, while this has allowed us to communicate to anywhere back home, it has some disadvantages. What you see there is the solution.`

> 			`	"There are fourteen relays between the wormhole and Earth, but that means routing through Elnath or Aldebaran. Those two relays are owned by Lovelace Labs, and they don't have enough bandwidth for the entirely of the Syndicate. During a backlog, information will be routed through even more indirect routes, which is holding up Syndicate business by problematic degrees."`

Author: The player is forced to read through lots of network jargon and will miss the plot points entirely. If you want them to get the plot points, then they must be presented more concisely than they were.

> 			`	"There are fourteen relays between the wormhole and Earth, but that means routing through Elnath or Aldebaran. Those two relays are owned by Lovelace Labs, and they don't have enough bandwidth for the entirely of the Syndicate. During a backlog, information will be routed through even more indirect routes, which is holding up Syndicate business by problematic degrees."`

Author: I made some corrections, and I think I make your plot points concisely now.

> 			`	"There are fourteen relays between the wormhole and Earth, but that means routing through Elnath or Aldebaran. Those two relays are owned by Lovelace Labs, and they don't have enough bandwidth for the entirely of the Syndicate. During a backlog, information will be routed through even more indirect routes, which is holding up Syndicate business by problematic degrees."`

Author: I just added an optional dialog that goes into the politics and economy of Lovelace Labs relays.

> 			`	"There are fourteen relays between the wormhole and Earth, but that means routing through Elnath or Aldebaran. Those two relays are owned by Lovelace Labs, and they don't have enough bandwidth for the entirely of the Syndicate. During a backlog, information will be routed through even more indirect routes, which is holding up Syndicate business by problematic degrees."`

Author: Okay, now the super-long paragraph about relay routes is also behind an optional dialog option, when the player unwisely suggests splitting the traffic across multiple routes.

			`	The emphasis he places on the words 'you' and 'helpful' suddenly give you the sense that this might be coordinated specifically for your benefit. Realizing this, you nod and accept it, on his account.`
			label accepted
			`	"Excellent," says Remington, standing up and flicking over the pickup details to your pad. "Check in with Terry on your way; she may have updated scouting information. Now I have another meeting to get to momentarily, but thank you, Captain. It's been a pleasure seeing you again."`
			`	He gives a meaningful look to Alondo on his way out, but exactly what it means is not clear to you. Alondo's posture relaxes as he gets up and quietly closes the door.`
			`	"Now," says Alondo, setting aside his pad. "We have some other business. I have a lead on this symbol you wanted investigating. I still wish you'd tell me why you're looking for it, but for now, I'll accept that it's important." he fishes around in a small satchel bag that usually carries his pad and pulls out an actual piece of physical paper. The text on it reads, 'Adjunct Research Library, Navy Intelligence Observatory, Moscow. Room number: 3017A19'.`

> 			`	"Fortunately, I found someone for you to speak to, and a cover for being there in the first place. Xilin needs a trusted aide to stay on top of things here, especially as Mountaintop progresses. So far he's found the staff provided there to be... lacking. I get the sense from him that he's expecting things to get more complicated, though he hasn't said in so many words. So, you get to go to Earth to pick up his assistant, and while you're there, you can speak to a certain Ms. Giti Gilani. She's an expert in the history of space flight and related archaic topics. Meanwhile the aide you'll pick up is a young man called Batu Cegeen. You'll have to land at the Ulaanbaatar spaceport for him, and travel locally for your business with Ms. Gilani."`

Reviewer 2: ```suggestion
			`	"Fortunately, I found someone for you to speak to and a cover for being there in the first place. Xilin needs a trusted aide to stay on top of things here, especially as Mountaintop progresses. So far he's found the staff provided there to be... lacking. I get the sense from him that he's expecting things to get more complicated, though he hasn't said in so many words. So, you get to go to Earth to pick up his assistant, and while you're there, you can speak to a certain Ms. Giti Gilani. She's an expert in the history of space flight and related archaic topics. Meanwhile the aide you'll pick up is a young man called Batu Cegeen. You'll have to land at the Ulaanbaatar spaceport for him, and travel locally for your business with Ms. Gilani."`
```

			label meeting
			`	"Ah, <last>, please come in." Remington moves over and clears your path to the nearest seat.`
			`	"It's good to see you back again, <last>," says Alondo, "and just in time for something that I think it's best to ask you to do. Would you like to tell the good Captain what the plan is, Remington?"`
			`	The slightly accentuated formality on 'Remington' is not lost on you, but the Syndicate PR man doesn't react to it as he hands you a datapad displaying a hyperspace relay. "Since this crisis began, the Navy maintained hyperspace relays in Rajak and Hasseleh, linking our communication network to the Hai's through a secure portal. However, while this has allowed us to communicate to anywhere back home, it has some disadvantages. What you see there is the solution.`
			`	"There are fourteen relays between the wormhole and Earth, but that means routing through Elnath or Aldebaran. Lovelace Labs owns those relays, and they get priority access, so our bandwidth is poor there. During a backlog, information will be routed through even more indirect routes, which is holding up Syndicate business by problematic degrees."`
			choice

> 				`	"And you want me to place some more relays?"`

Reviewer 3: ```suggestion
				`	"And you want me to place some new relays?"`
```

					goto relays
				`	"Can't you just change the priorities of the Lovelace Labs relays?"`
					goto lovenot
				`	"What if you distribute the traffic over multiple routes?"`
					goto multiple
				`	"So why haven't you placed more relays?"`
					goto more
			label lovenot

> 			`	Remington says, "They're privately-owned relays, so we have no say over them. Lovelace Labs needs that bandwidth for their internal purposes, so they can't afford to lose priority for their own traffic. We pay a stiff sum to get the highest priority they are willing to part with."`

Reviewer 3: Felt like this communicated it a bit clearer.
```suggestion
			`	Remington says, "They're privately-owned relays, so we have no say over them. Lovelace Labs reserves priority status on that bandwidth for their internal purposes. We pay a stiff sum to get the highest priority they are willing to part with, but often they don't leave much for anyone else."`
```

				`	"I thought even pirates wouldn't blow up relays?"`
				`	"Why would they be doing that, don't they use them too?"`
			label relays

> 			`	Remington nods. "Yes, well, this end of space has never had relays since nobody needed them. For a long time the marauders here have been using the communications darkness out this way to do their business with impunity. The sole advantage of these relays is that ships out this way can call for help when they couldn't before. So the relays the Syndicate have been putting up have been getting destroyed, along with our security ships.`

Reviewer: ```suggestion
			`	Remington nods. "Yes, well, this end of space has never had relays since nobody needed them. For a long time the marauders here have been using the communications darkness out this way to do their business with impunity. The sole advantage of these relays is that ships out this way can call for help when they couldn't before. So the relays the Syndicate have been putting up have been getting destroyed, along with several of our security ships.`
```

I want to make clear that the security ships were related to the relays, since Syndicate security ships get blown up all the time.

> 			`	"This one that you see in front of you is different though." He makes eye contact with Alondo and very deliberately emphases the word as he continues, "WE have been working with the Hai to make relays that look human-made, but utilize advanced Hai technology to baffle scanners attempting to find them. They'll form communication links the marauders can't destroy."`

Reviewer: ```suggestion
			`	"This one that you see in front of you is different, though." He makes eye contact with Alondo and very deliberately emphases the word as he continues, "WE have been working with the Hai to make relays that look human-made, but utilize advanced Hai technology to baffle scanners attempting to find them. They'll form communication links the marauders can't destroy."`
```

			`	He looks at you, "But of course, since they have Hai tech in them, they need to be delivered a trustworthy human who's not likely to get blown up by a marauder."`
			choice
				`	"I can deploy the relays for you."`
					goto accepted

> 				`	"Not sure I want to do this."`

Reviewer: I'm now wondering if this next bit to refuse the offer isn't useful at all. Refusing it only to make the player accept it doesn't feel right.

Author: I agree, but I didn't add that content, and it doesn't get in the way of anything, so I left it there.

A lot of critical missions in Endless Sky let you refuse to help even though it doesn't make sense to do so. This one is nice enough to not cut you out of the rest of the game for that.

			`	Alondo looks at you and says, "We need you to do this. It'll be very helpful."`
			`	The emphasis he places on the words 'you' and 'helpful' suddenly give you the sense that this might be coordinated specifically for your benefit. Realizing this, you nod and accept it, on his account.`
			label accepted
			`	"Excellent," says Remington, standing up and flicking over the pickup details to your pad. "Check in with Terry on your way; she may have updated scouting information. Now I have another meeting to get to momentarily, but thank you, Captain. It's been a pleasure seeing you again."`
			`	He gives a meaningful look to Alondo on his way out, but exactly what it means is not clear to you. Alondo's posture relaxes as he gets up and quietly closes the door.`
			`	"Now on to our business. I have a lead on this symbol you found. I still wish you'd tell me why you're looking for it, but for now, I'll accept that it's important." he fishes around in a small satchel bag that usually carries his pad and pulls out an actual piece of physical paper. The text on it reads, 'Adjunct Research Library, Navy Intelligence Observatory, Moscow. Room number: 3017A19'.`

> 			`	"Fortunately, I found someone for you to speak to AND a cover for being there in the first place. Xilin needs a trusted aide to stay on top of things here, especially as Mountaintop progresses. So far he's found the staff provided to be... lacking. I get the sense from him that he's expecting things to get more complicated, though he hasn't said in so many words. So, you get to go to Earth to pick up his assistant, and while there, you can speak to a certain Ms. Giti Gilani. She's an expert in the history of space flight and related archaic topics. Meanwhile the aide you'll pick up is a young man called Batu Cegeen. You'll have to land at the Ulaanbaatar spaceport for him and travel locally for your business with Ms. Gilani."`

Reviewer: ```suggestion
			`	"Fortunately, I found someone for you to speak to, and a cover for being there in the first place. Xilin needs a trusted aide to stay on top of things here, especially as Mountaintop progresses. So far he's found the staff provided there to be... lacking. I get the sense from him that he's expecting things to get more complicated, though he hasn't said in so many words. So, you get to go to Earth to pick up his assistant, and while you're there, you can speak to a certain Ms. Giti Gilani. She's an expert in the history of space flight and related archaic topics. Meanwhile the aide you'll pick up is a young man called Batu Cegeen. You'll have to land at the Ulaanbaatar spaceport for him, and travel locally for your business with Ms. Gilani."`
```

			label regular
			`	"Yes, my regular job. Specifically the one I had before the end of the war, and before this farce they have me doing now, while my successor tries and fails to be half as competent."`
			label card
			`	She pushes a button her datapad's case and it ejects a business card which she hands to you. You see on the card, "Chief Operations Officer".`
			`	"Of course," she says, taking the card back and neatly pushing into the case, "I'm not allowed to dissemate those cards at the moment. At least not until this is over, when I can take back my position after the fool in my place has made a mess of things."`
			`	She seems absolutely sure that that's exactly how things are going to turn out.`
			`	"Now, speaking of my second job, you have relays on your ship that will make that easier, yes?"`
			choice
				`	"Yes, I have the relays."`

> 			`	"Wonderful, the latency has been killing me lately. These are the coordinates for deployment." She flicks the relevant data files from her pad to you. "Make sure those are loaded into the relays, deploy roughly the right place, and they'll move to their own positions."`

Reviewer: ```suggestion
			`	"Wonderful, the latency has been killing me lately. These are the coordinates for deployment." She flicks the relevant data files from her pad to you. "Make sure those are loaded into the relays. Deploy them roughly in the right place, and they'll move to their own positions."`
```

			label downplay
			`	"No need to downplay yourself, Captain! You've done God's work in saving humanity from that alien scourge.`
			label introduction

> 			`	Terry looks between you. "Captain <last> is doing a lot to help us out, so please, make yourself acquainted. Don't too long though, the Captain has a job to do." With that, she gives the both of you a parting nod and strides back into the chaos. Almost immediately a swarm of people with datapads materialize from nowhere and pursue her.`

Reviewer: ```suggestion
			`	Terry looks between you. "Captain <last> is doing a lot to help us out, so please make yourselves acquainted. Don't be too long though, the Captain has a job to do." With that, she gives the both of you a parting nod and strides back into the chaos. Almost immediately a swarm of people with datapads materialize from nowhere and pursue her.`
```

			`	The man grins and shakes his head as she departs. "Remarkable woman. Anyway, the name's David Joseph Turner, venture capitalist, entrepreneur, and famous Paradise Worlds businessman." Turner grabs a business card from his pocket and hands it to you. It contains his contact information, an address on Martini, and boasts in golden letters his net worth of more than three billion credits.`
			`	"You may not have heard of me, but I've sure as hell heard of you." Turner turns around and waves his arm across the crowd of people working on the station concourse. "To be involved in funding and producing this movie, though, what an adventure. You know I've paid for movies to be made, or at least invested in them, but I've never actually been there while it was happening. Quite remarkable."`

> 			`	"Of course, there are the communication latency issues, and the dreadful matter of the black spot that is the route through Cardax. We need proper coverage of that area if we're going to provide security to the tourists who seem intent on coming here. Though, I hear that you might be working on that as we speak am I right?"`

Reviewer: ```suggestion
			`	"Of course, there are the communication latency issues, and the dreadful matter of the black spot that is the route through Cardax. We need proper coverage of that area if we're going to provide security to the tourists who seem intent on coming here. Though I hear that you might be working on that as we speak, am I right?"`
```

			choice
				`	"Yes, I'll fix the latency issue."`
					goto yes

> 				`	"Those systems are on my way."`

Author: The "yes" was confusing. I just reverted "It's on my way." and changed "Yes." to "Yes, I'll fix it."
